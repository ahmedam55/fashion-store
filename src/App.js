import { QueryClientProvider, QueryClient } from 'react-query'
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom'

import ProductDetails from './components/ProductDetails'

import './style.scss'

const queryClient = new QueryClient({
  defaultOptions: { queries: { refetchOnWindowFocus: false } },
})

function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <Router>
        <Route path="/:id?" component={ProductDetails}></Route>
      </Router>
    </QueryClientProvider>
  )
}

export default App
