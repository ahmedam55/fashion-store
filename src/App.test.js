import { render, screen } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer';
import App from './App'


it('Jest Snapshot Matching: Whole app renders properly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})


it('Loading shimmer is showing initially', () => {
  render(<App />)

  const loadingShimmer = screen.getByText(/Loading.../i)
  expect(loadingShimmer).toBeInTheDocument()
})


it('Nav Menu is showing properly', () => {
  render(<App />)

  const navMenu = screen.getAllByText(/Men|Women|Kids|FTW|Accessories/i)

  navMenu.map(menuItem => expect(menuItem).toBeInTheDocument())
})
