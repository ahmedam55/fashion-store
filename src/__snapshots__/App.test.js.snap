// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot Matching: Whole app renders properly 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <div
      className="logoWrapper"
    >
      <img
        alt="logo"
        className="logo"
        src="/images/logo.jpg"
      />
    </div>
    <ul
      className="navMenu"
    >
      <li
        className="navMenuItem"
      >
        <a
          href="#"
        >
          Men
        </a>
      </li>
      <li
        className="navMenuItem"
      >
        <a
          href="#"
        >
          Women
        </a>
      </li>
      <li
        className="navMenuItem"
      >
        <a
          href="#"
        >
          Kids
        </a>
      </li>
      <li
        className="navMenuItem"
      >
        <a
          href="#"
        >
          FTW
        </a>
      </li>
      <li
        className="navMenuItem"
      >
        <a
          href="#"
        >
          Accessories
        </a>
      </li>
    </ul>
    <div
      className="searchWrapper"
    >
      <input
        className="search"
        placeholder="Search..."
      />
      <svg
        className="searchIcon"
        viewBox="0 0 24 24"
      >
        <g
          data-name="Layer 2"
        >
          <path
            d="M20.71 19.29l-3.4-3.39A7.92 7.92 0 0019 11a8 8 0 10-8 8 7.92 7.92 0 004.9-1.69l3.39 3.4a1 1 0 001.42 0 1 1 0 000-1.42zM5 11a6 6 0 116 6 6 6 0 01-6-6z"
            data-name="search"
          />
        </g>
      </svg>
    </div>
    <svg
      aria-labelledby="2wl92si-aria"
      className="loading"
      height={905}
      id={null}
      role="img"
      style={Object {}}
      viewBox="0 0 605 905"
      width={605}
    >
      <title
        id="2wl92si-aria"
      >
        Loading...
      </title>
      <rect
        clipPath="url(#2wl92si-diff)"
        height="100%"
        role="presentation"
        style={
          Object {
            "fill": "url(#2wl92si-animated-diff)",
          }
        }
        width="100%"
        x="0"
        y="0"
      />
      <defs
        role="presentation"
      >
        <clipPath
          id="2wl92si-diff"
        >
          <rect
            height="432"
            rx="5"
            ry="5"
            width="331"
            x="133"
            y="92"
          />
          <rect
            height="16"
            rx="0"
            ry="0"
            width="166"
            x="12"
            y="599"
          />
          <rect
            height="12"
            rx="0"
            ry="0"
            width="87"
            x="12"
            y="634"
          />
          <rect
            height="12"
            rx="0"
            ry="0"
            width="259"
            x="12"
            y="678"
          />
          <rect
            height="10"
            rx="0"
            ry="0"
            width="80"
            x="12"
            y="724"
          />
          <rect
            height="28"
            rx="5"
            ry="5"
            width="34"
            x="12"
            y="749"
          />
          <rect
            height="28"
            rx="5"
            ry="5"
            width="34"
            x="54"
            y="749"
          />
          <rect
            height="28"
            rx="5"
            ry="5"
            width="34"
            x="96"
            y="749"
          />
          <rect
            height="28"
            rx="5"
            ry="5"
            width="34"
            x="138"
            y="749"
          />
          <rect
            height="13"
            rx="0"
            ry="0"
            width="96"
            x="12"
            y="828"
          />
          <rect
            height="10"
            rx="0"
            ry="0"
            width="543"
            x="12"
            y="863"
          />
          <rect
            height="11"
            rx="0"
            ry="0"
            width="543"
            x="12"
            y="889"
          />
        </clipPath>
        <linearGradient
          id="2wl92si-animated-diff"
        >
          <stop
            offset="0%"
            stopColor="#f3f3f3"
            stopOpacity={1}
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-2; -2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stopColor="#ecebeb"
            stopOpacity={1}
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-1; -1; 2"
            />
          </stop>
          <stop
            offset="100%"
            stopColor="#f3f3f3"
            stopOpacity={1}
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="0; 0; 3"
            />
          </stop>
        </linearGradient>
      </defs>
    </svg>
  </div>
</div>
`;
